### DISCUSSION TITLE
Building a real-time chat in Gleam

### DISCUSSION THREAD

**Discussion by giscus** [GauBen/gautier.dev#6] [Comments]
# Building a real-time chat in Gleam

Elegant, safe and pragmatic, I have found my favorite language: Gleam. Let's build a real-time chat application with it!

https://gautier.dev/articles/real-time-gleam-chat

**Comment by giovannibonetti** [comment]
Great write-up, very clear and practical

**Reply by GauBen** [reply]
Thank you!

**Comment by mtsknn** [comment]
Very nice article! And the final code is clear and short. 👏

At first I was confused with `->` in some of the code snippets; I thought "Gleam doesn't have that kind of arrow functions!" Later I realized that `->` is used in case expressions. 😄

> As a matter of fact, SSE are usually a better choice than WebSockets for simple use cases, as they are easier to implement and some features like reconnection are built in.

I'm not familiar with SSEs so this was interesting. I was going to ask if you have pointers for learning more about SSEs vs WebSockets, but well, maybe I should just try googling.

I have a question about [the final code on GitHub (lines 85–92)](https://github.com/GauBen/real-time-gleam-chat/blob/ccb192136c64eb7f7fcc7c89a7dc94374bafba32/src/real_time_chat.gleam#L85-L92): in `POST /post`, why do you read only the first 128 bytes of the request?

**Reply by GauBen** [reply]
Thank you very much!

I love SSE, I think WebSockets are overkill for most use cases, so I'm glad you discovered a lightweight alternative.

> why do you read only the first 128 bytes of the request?

I limited the input field with [`maxlength="128"`](https://github.com/GauBen/real-time-gleam-chat/blob/main/src/index.html#L59) in the frontend, so no legitimate message over this limit should reach the server. I could have pulled the entire request body into RAM, but I consider that a bad security practice, a malicious actor could easily bring down the server. Even if the resulting code is not production-ready, I try not to publish vulnerable code.

**Reply by mtsknn** [reply]
> I limited the input field with `maxlength="128"` in the frontend, so no legitimate message over this limit should reach the server.

Ah, makes sense. I had missed the `maxlength` attribute. (A code comment in the `.gleam` file might be in order.)

> I could have pulled the entire request body into RAM, but I consider that a bad security practice, a malicious actor could easily bring down the server. Even if the resulting code is not production-ready, I try not to publish vulnerable code.

Good points. 👍

**Reply by aemonge** [reply]
This Gist is gold for me, thanks ! <3
